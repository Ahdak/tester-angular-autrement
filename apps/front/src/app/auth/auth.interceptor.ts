import { Injectable } from '@angular/core';
import {
  HttpInterceptor,
  HttpRequest,
  HttpHandler,
} from '@angular/common/http';
import { mergeMap, take } from 'rxjs/operators';

import { AuthFacade } from './+state/auth.facade';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  static readonly tokenHeaderName = 'x-auth-token';

  constructor(private authFacade: AuthFacade) {}

  intercept(req: HttpRequest<any>, next: HttpHandler) {
    return this.authFacade.accessToken.pipe(
      take(1),
      mergeMap(accessToken => {
        const authReq = req.clone(
          accessToken
            ? {
                headers: req.headers.set(
                  AuthInterceptor.tokenHeaderName,
                  accessToken,
                ),
              }
            : {},
        );
        return next.handle(authReq);
      }),
    );
  }
}
