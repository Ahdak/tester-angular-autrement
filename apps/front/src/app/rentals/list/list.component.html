<div class="grid-container">
  <div fxLayout="row"
       fxLayoutAlign="space-between center">
    <h1 class="mat-h1">Logements disponibles</h1>
    <mat-form-field fxFlex="200px" class="search-form">
      <input
        [formControl]="searchControl"
        matInput
        type="search" placeholder="Chercher une ville"
      />
      <button
        *ngIf="searchControl.value"
        (click)="searchControl.setValue('')"
        mat-button
        matSuffix mat-icon-button aria-label="Clear">
        <mat-icon>close</mat-icon>
      </button>
    </mat-form-field>
  </div>
  <div *ngIf="loading$ | async; else rentalsList" fxLayout="row" fxLayoutAlign="center center">
    <mat-progress-spinner mode="indeterminate"></mat-progress-spinner>
  </div>
  <ng-template #rentalsList>
    <mat-grid-list cols="3">
      <mat-grid-tile *ngFor="let rental of rentals$ | async" [colspan]="1" [rowspan]="1">
        <mat-card class="home-card" [attr.data-testid]="'rental-'  + rental.id">
          <mat-card-header>
            <mat-card-title>
              <a [routerLink]="['/rentals/', rental.id]" data-testid="title">{{rental.title}}</a>
            </mat-card-title>
          </mat-card-header>
          <img [routerLink]="['/rentals/', rental.id]" mat-card-image [height]="190" [src]="rental.image" data-testid="image">
          <mat-card-content>
            <mat-chip-list>
              <mat-chip color="primary" selected data-testid="city">{{rental.city}}</mat-chip>
              <mat-chip data-testid="type">{{rental.type}}</mat-chip>
              <mat-chip data-testid="roomsNbr">{{rental.roomsNbr}} Ch</mat-chip>
              <mat-chip data-testid="guestsNbr">{{rental.guestsNbr}} Pers</mat-chip>
              <mat-chip *ngIf="loggedIn$ | async; else logInForPrice" color="accent" selected data-testid="price">{{rental.price}} € par nuit</mat-chip>
              <ng-template #logInForPrice>
                <mat-chip color="warn" selected data-testid="price">Connectez-vous pour voir le prix</mat-chip>
              </ng-template>
            </mat-chip-list>
            <div class="actions">
              <button [routerLink]="['/rentals/', rental.id]" mat-stroked-button color="accent" data-testid="book">Réserver</button>
            </div>
          </mat-card-content>
        </mat-card>
      </mat-grid-tile>
    </mat-grid-list>
  </ng-template>
</div>
