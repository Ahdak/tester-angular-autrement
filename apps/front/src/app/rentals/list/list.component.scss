.grid-container {
  margin: 20px;
}

.home-card {
  position: absolute;
  top: 15px;
  left: 15px;
  right: 15px;
  bottom: 15px;
}

mat-card-title {
  font-weight: 700;
  font-size: 15px;
  letter-spacing: 0.2px;

  a {
    text-decoration: none;
    color: rgb(72, 72, 72);
  }
}

[mat-card-image]:hover {
  cursor: pointer;
}

mat-card-content {
  line-height: 18px;
  letter-spacing: 0.2px;
}

.actions {
  margin-top: 10px;
}
//mat-card-actions {
//  button {
//    margin-left: 15px;
//  }
//}
