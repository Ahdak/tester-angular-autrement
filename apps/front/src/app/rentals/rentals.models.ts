export interface Rental {
  id: number;
  title: string;
  description: string;
  city: string;
  type: Type;
  guestsNbr: number;
  roomsNbr: number;
  price: number;
  image: string;
}

export enum Type {
  ENTIRE_HOME = 'Logement entier',
  PRIVATE_ROOM = 'Chambre privée',
  SHARED_ROOM = 'Chambre partagée',
}

export interface RentalsQuery {
  city?: string;
}
