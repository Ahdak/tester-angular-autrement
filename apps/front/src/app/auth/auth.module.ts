import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { authReducers } from './+state/auth.selectors';
import { AuthEffects } from './+state/auth.effects';
import { LoginFormComponent } from './login-form/login-form.component';
import { LoginPageComponent } from './login-page/login-page.component';

@NgModule({
  imports: [
    SharedModule,
    StoreModule.forFeature('auth', authReducers),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: [LoginFormComponent, LoginPageComponent],
})
export class AuthModule {}
