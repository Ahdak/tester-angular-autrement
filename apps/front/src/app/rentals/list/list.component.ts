import { Component, OnDestroy, OnInit } from '@angular/core';
import { RentalsFacade } from '../+state/rentals.facade';
import { AuthFacade } from '../../auth/+state/auth.facade';
import { FormControl } from '@angular/forms';
import {
  debounceTime,
  distinctUntilChanged,
  startWith,
  takeUntil,
} from 'rxjs/operators';
import { Subject } from 'rxjs';

@Component({
  selector: 'app-list',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.scss'],
})
export class ListComponent implements OnInit, OnDestroy {
  loading$ = this.rentalsFacade.loading$;
  rentals$ = this.rentalsFacade.allRentals$;
  loggedIn$ = this.authFacade.loggedIn$;

  searchControl = new FormControl();
  destroy$ = new Subject();

  constructor(
    private rentalsFacade: RentalsFacade,
    private authFacade: AuthFacade,
  ) {}

  ngOnInit(): void {
    this.searchControl.valueChanges
      .pipe(
        debounceTime(300),
        distinctUntilChanged(),
        takeUntil(this.destroy$),
        startWith(''),
      )
      .subscribe(city => {
        this.rentalsFacade.loadAll({ city });
      });
  }

  ngOnDestroy(): void {
    this.destroy$.next();
  }
}
