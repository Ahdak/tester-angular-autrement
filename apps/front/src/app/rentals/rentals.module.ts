import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '../shared/shared.module';
import { ListComponent } from './list/list.component';
import { DetailsComponent } from './details/details.component';
import { initialState as rentalsInitialState, rentalsReducer } from './+state/rentals.reducer';
import { RentalsEffects } from './+state/rentals.effects';
import { BookingModule } from '../booking/booking.module';

@NgModule({
  imports: [
    SharedModule,
    BookingModule,
    StoreModule.forFeature('rentals', rentalsReducer, {
      initialState: rentalsInitialState,
    }),
    EffectsModule.forFeature([RentalsEffects]),
  ],
  declarations: [ListComponent, DetailsComponent],

})
export class RentalsModule {}
