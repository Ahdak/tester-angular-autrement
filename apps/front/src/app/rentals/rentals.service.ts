import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { delay } from 'rxjs/operators';

import { Rental, RentalsQuery } from './rentals.models';

@Injectable({
  providedIn: 'root',
})
export class RentalsService {
  constructor(private http: HttpClient) {}

  getRentals(query: RentalsQuery): Observable<Rental[]> {
    let params;
    if (query.city) {
      params = new HttpParams();
      params = params.append('city', query.city);
    }
    return this.http
      .get<Rental[]>('/api/rentals', params ? { params } : undefined)
      .pipe(delay(1000));
  }

  getRental(id: number): Observable<Rental> {
    return this.http.get<Rental>(`/api/rentals/${id}`).pipe(delay(1000));
  }
}
