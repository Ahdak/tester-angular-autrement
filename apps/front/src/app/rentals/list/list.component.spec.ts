import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { Router } from '@angular/router';
import { HttpClient } from '@angular/common/http';
import { of } from 'rxjs';

import { ListComponent } from './list.component';
import { SharedModule } from '../../shared/shared.module';
import { rentalsReducer } from '../+state/rentals.reducer';
import { RentalsEffects } from '../+state/rentals.effects';
import { authReducers } from '../../auth/+state/auth.selectors';
import { Rental, Type } from '../rentals.models';
import { LoginSuccess } from '../../auth/+state/auth.actions';


describe('TestComponent', () => {
  let component: ListComponent;
  let fixture: ComponentFixture<ListComponent>;
  let nativeElement: Element;

  let router: Router;
  let httpClient: HttpClient;
  let store: Store<any>;

  const rentals: Rental[] = [
    {
      id: 1, title: 'Title', price: 80, description: 'desc', city: 'Paris',
      image: 'image', type: Type.ENTIRE_HOME, roomsNbr: 2, guestsNbr: 4
    }
  ];

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ListComponent ],
      imports: [
        StoreModule.forRoot({
          'auth': combineReducers(authReducers),
          'rentals': rentalsReducer,
        }),
        EffectsModule.forRoot([RentalsEffects]),
        SharedModule,
        NoopAnimationsModule,
        HttpClientTestingModule,
        RouterTestingModule,
      ],
      providers: [
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListComponent);
    component = fixture.componentInstance;
    nativeElement = fixture.nativeElement;

    router = TestBed.get(Router);
    httpClient = TestBed.get(HttpClient);
    store = TestBed.get(Store);

    spyOn(router, 'navigate').and.stub();
    spyOn(httpClient, 'get').and.returnValue(of(rentals));
    fixture.detectChanges();
  });

  it('should create instance', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot when not logged in', () => {
    expect(httpClient.get).toHaveBeenCalledWith('/api/rentals', undefined);
    expect(nativeElement).toMatchSnapshot();
  });

  it('should match snapshot when logged in', () => {
    store.dispatch(
      new LoginSuccess({user: {username: 'user', firstname: 'john', lastname: 'doe', isPremium: true}, accessToken: 'token'})
    );
    fixture.detectChanges();
    fixture.whenStable().then(() => {
      expect(nativeElement).toMatchSnapshot();
    })
  });
});
