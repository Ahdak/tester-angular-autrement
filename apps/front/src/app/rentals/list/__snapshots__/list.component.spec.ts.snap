// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should match snapshot when logged in 1`] = `
<div
  id="root2"
  ng-version="6.1.10"
>
  <div
    class="grid-container"
  >
    <div
      fxlayout="row"
      fxlayoutalign="space-between center"
      ng-reflect-align="space-between center"
      ng-reflect-layout="row"
      style="box-sizing: border-box; display: flex; flex-direction: row; max-height: 100%; align-content: center; align-items: center; justify-content: space-between;"
    >
      <h1
        class="mat-h1"
      >
        Logements disponibles
      </h1>
      <mat-form-field
        class="search-form mat-form-field ng-tns-c16-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
        fxflex="200px"
        ng-reflect-flex="200px"
        style="box-sizing: border-box; max-width: 200px; min-width: 200px;"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
            <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
            <div
              class="mat-form-field-infix"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                id="mat-input-2"
                matinput=""
                ng-reflect-form="[object Object]"
                ng-reflect-placeholder="Chercher une ville"
                ng-reflect-type="search"
                placeholder="Chercher une ville"
                type="search"
              />
              <span
                class="mat-form-field-label-wrapper"
              >
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <label
                  aria-owns="mat-input-2"
                  class="mat-form-field-label ng-tns-c16-0 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-input-2"
                  id="mat-form-field-label-5"
                  ng-reflect-ng-switch="false"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                  <!---->
                  Chercher une ville
                  <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                </label>
              </span>
            </div>
            <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
          </div>
          <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
          <div
            class="mat-form-field-underline ng-tns-c16-0 ng-star-inserted"
          >
            <span
              class="mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
            <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
            <div
              class="mat-form-field-hint-wrapper ng-tns-c16-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              <!--bindings={
  "ng-reflect-ng-if": ""
}-->
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
    </div>
    <!--bindings={
  "ng-reflect-ng-if": "false",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
    <mat-grid-list
      class="mat-grid-list ng-star-inserted"
      cols="3"
      ng-reflect-cols="3"
    >
      <div>
        <!--bindings={
  "ng-reflect-ng-for-of": "[object Object]"
}-->
        <mat-grid-tile
          class="mat-grid-tile ng-star-inserted"
          ng-reflect-colspan="1"
          ng-reflect-rowspan="1"
          style="left: 0px; margin-top: 0px;"
        >
          <figure
            class="mat-figure"
          >
            <mat-card
              class="home-card mat-card"
            >
              <mat-card-header
                class="mat-card-header"
              >
                <div
                  class="mat-card-header-text"
                >
                  <mat-card-title
                    class="mat-card-title"
                  >
                    <a
                      href="/rentals/1"
                      ng-reflect-router-link="/rentals/,1"
                    >
                      Title
                    </a>
                  </mat-card-title>
                </div>
              </mat-card-header>
              <img
                class="mat-card-image"
                height="190"
                mat-card-image=""
                ng-reflect-router-link="/rentals/,1"
                src="image"
                tabindex="0"
              />
              <mat-card-content
                class="mat-card-content"
              >
                <mat-chip-list
                  aria-disabled="false"
                  aria-invalid="false"
                  aria-multiselectable="false"
                  aria-orientation="horizontal"
                  aria-required="false"
                  class="mat-chip-list"
                  id="mat-chip-list-2"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    class="mat-chip-list-wrapper"
                  >
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="true"
                      class="mat-chip mat-primary mat-standard-chip mat-chip-selected"
                      color="primary"
                      ng-reflect-color="primary"
                      ng-reflect-selected=""
                      role="option"
                      selected=""
                      tabindex="-1"
                    >
                      Paris
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      Logement entier
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      2 Ch
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      4 Pers
                    </mat-chip>
                    <!--bindings={
  "ng-reflect-ng-if": "true",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="true"
                      class="mat-chip mat-standard-chip mat-accent mat-chip-selected ng-star-inserted"
                      color="accent"
                      ng-reflect-color="accent"
                      ng-reflect-selected=""
                      role="option"
                      selected=""
                      tabindex="-1"
                    >
                      80 € par nuit
                    </mat-chip>
                    <!---->
                  </div>
                </mat-chip-list>
                <div
                  class="actions"
                >
                  <button
                    class="mat-stroked-button mat-accent _mat-animation-noopable"
                    color="accent"
                    mat-stroked-button=""
                    ng-reflect-color="accent"
                    ng-reflect-router-link="/rentals/,1"
                    tabindex="0"
                  >
                    <span
                      class="mat-button-wrapper"
                    >
                      Réserver
                    </span>
                    <div
                      class="mat-button-ripple mat-ripple"
                      matripple=""
                      ng-reflect-centered="false"
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="[object HTMLButtonElement]"
                    />
                    <div
                      class="mat-button-focus-overlay"
                    />
                  </button>
                </div>
              </mat-card-content>
            </mat-card>
          </figure>
        </mat-grid-tile>
      </div>
    </mat-grid-list>
    <!---->
  </div>
</div>
`;

exports[`TestComponent should match snapshot when not logged in 1`] = `
<div
  id="root1"
  ng-version="6.1.10"
>
  <div
    class="grid-container"
  >
    <div
      fxlayout="row"
      fxlayoutalign="space-between center"
      ng-reflect-align="space-between center"
      ng-reflect-layout="row"
      style="box-sizing: border-box; display: flex; flex-direction: row; max-height: 100%; align-content: center; align-items: center; justify-content: space-between;"
    >
      <h1
        class="mat-h1"
      >
        Logements disponibles
      </h1>
      <mat-form-field
        class="search-form mat-form-field ng-tns-c8-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
        fxflex="200px"
        ng-reflect-flex="200px"
        style="box-sizing: border-box; max-width: 200px; min-width: 200px;"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
            <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
            <div
              class="mat-form-field-infix"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                id="mat-input-1"
                matinput=""
                ng-reflect-form="[object Object]"
                ng-reflect-placeholder="Chercher une ville"
                ng-reflect-type="search"
                placeholder="Chercher une ville"
                type="search"
              />
              <span
                class="mat-form-field-label-wrapper"
              >
                <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
                <label
                  aria-owns="mat-input-1"
                  class="mat-form-field-label ng-tns-c8-0 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-input-1"
                  id="mat-form-field-label-3"
                  ng-reflect-ng-switch="false"
                >
                  <!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                  <!---->
                  Chercher une ville
                  <!--bindings={
  "ng-reflect-ng-switch-case": "true"
}-->
                  <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
                </label>
              </span>
            </div>
            <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
          </div>
          <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
          <div
            class="mat-form-field-underline ng-tns-c8-0 ng-star-inserted"
          >
            <span
              class="mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
            <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
            <div
              class="mat-form-field-hint-wrapper ng-tns-c8-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              <!--bindings={
  "ng-reflect-ng-if": ""
}-->
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
    </div>
    <!--bindings={
  "ng-reflect-ng-if": "false",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
    <mat-grid-list
      class="mat-grid-list ng-star-inserted"
      cols="3"
      ng-reflect-cols="3"
    >
      <div>
        <!--bindings={
  "ng-reflect-ng-for-of": "[object Object]"
}-->
        <mat-grid-tile
          class="mat-grid-tile ng-star-inserted"
          ng-reflect-colspan="1"
          ng-reflect-rowspan="1"
          style="left: 0px; margin-top: 0px;"
        >
          <figure
            class="mat-figure"
          >
            <mat-card
              class="home-card mat-card"
            >
              <mat-card-header
                class="mat-card-header"
              >
                <div
                  class="mat-card-header-text"
                >
                  <mat-card-title
                    class="mat-card-title"
                  >
                    <a
                      href="/rentals/1"
                      ng-reflect-router-link="/rentals/,1"
                    >
                      Title
                    </a>
                  </mat-card-title>
                </div>
              </mat-card-header>
              <img
                class="mat-card-image"
                height="190"
                mat-card-image=""
                ng-reflect-router-link="/rentals/,1"
                src="image"
                tabindex="0"
              />
              <mat-card-content
                class="mat-card-content"
              >
                <mat-chip-list
                  aria-disabled="false"
                  aria-invalid="false"
                  aria-multiselectable="false"
                  aria-orientation="horizontal"
                  aria-required="false"
                  class="mat-chip-list"
                  id="mat-chip-list-1"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    class="mat-chip-list-wrapper"
                  >
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="true"
                      class="mat-chip mat-primary mat-standard-chip mat-chip-selected"
                      color="primary"
                      ng-reflect-color="primary"
                      ng-reflect-selected=""
                      role="option"
                      selected=""
                      tabindex="-1"
                    >
                      Paris
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      Logement entier
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      2 Ch
                    </mat-chip>
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="false"
                      class="mat-chip mat-primary mat-standard-chip"
                      role="option"
                      tabindex="-1"
                    >
                      4 Pers
                    </mat-chip>
                    <!--bindings={
  "ng-reflect-ng-if": "false",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
                    <mat-chip
                      aria-disabled="false"
                      aria-selected="true"
                      class="mat-chip mat-standard-chip mat-warn mat-chip-selected ng-star-inserted"
                      color="warn"
                      ng-reflect-color="warn"
                      ng-reflect-selected=""
                      role="option"
                      selected=""
                      tabindex="-1"
                    >
                      Connectez-vous pour voir le prix
                    </mat-chip>
                    <!---->
                  </div>
                </mat-chip-list>
                <div
                  class="actions"
                >
                  <button
                    class="mat-stroked-button mat-accent _mat-animation-noopable"
                    color="accent"
                    mat-stroked-button=""
                    ng-reflect-color="accent"
                    ng-reflect-router-link="/rentals/,1"
                    tabindex="0"
                  >
                    <span
                      class="mat-button-wrapper"
                    >
                      Réserver
                    </span>
                    <div
                      class="mat-button-ripple mat-ripple"
                      matripple=""
                      ng-reflect-centered="false"
                      ng-reflect-disabled="false"
                      ng-reflect-trigger="[object HTMLButtonElement]"
                    />
                    <div
                      class="mat-button-focus-overlay"
                    />
                  </button>
                </div>
              </mat-card-content>
            </mat-card>
          </figure>
        </mat-grid-tile>
      </div>
    </mat-grid-list>
    <!---->
  </div>
</div>
`;
